### PR TITLE
Added parseNaclResult to workspace API

### PR DESCRIPTION
**Pull request by amitgoldarb**

### CODE REVIEW

File: packages/workspace/src/workspace/nacl_files/nacl_files_source.ts [hunk 1]
```diff
@@ -369,10 +371,22 @@ const buildNaclFilesSource = (
...
-    getElements: async filename =>
-      Object.values((await getState()).parsedNaclFiles[filename]?.elements) || [],
+    getParsedNaclFile: async filename => {
+      // We don't want to parse all nacl files here when we want only parsedResult of one file.
+      if (state) {
+        return (await getState()).parsedNaclFiles[filename]
+      }
+      const naclFile = await getNaclFile(filename)
+      if (!naclFile) return undefined
+      const parseResult = await parse(
```
Comment (ori-moisis) [inline packages/workspace/src/workspace/nacl_files/nacl_files_source.ts:383]: not using the cache here might be a problem if the point is to improve performance in the short term

File: packages/workspace/test/common/nacl_file_source.ts [hunk 2]
```diff
@@ -56,8 +56,8 @@ export const createMockNaclFileSource = (
       end: {},
     }))),
   getErrors: jest.fn().mockImplementation(() => Promise.resolve(errors)),
-  getElements: jest.fn().mockImplementation(
-    filename => Promise.resolve(naclFiles[filename] || [])
+  getParsedNaclFile: jest.fn().mockImplementation(
+    filename => Promise.resolve({ elements: naclFiles[filename] || [] })
```
Comment (ori-moisis) [inline packages/workspace/test/common/nacl_file_source.ts:60]: seems like this does not implement the correct interface (parsed nacl result has more required fields)

File: packages/workspace/test/workspace/nacl_files/nacl_files_source.test.ts [hunk 3]
```diff
@@ -194,7 +194,29 @@ describe('Nacl Files Source', () => {
...
+  describe('getParsedNaclFile', () => {
+    let naclSource: NaclFilesSource
+
+    beforeEach(() => {
+      naclSource = naclFilesSource(
+        mockDirStore, mockCache, mockedStaticFilesSource
+      )
+    })
+
+    it('should return undefined', async () => {
```
Comment (ori-moisis) [inline packages/workspace/test/workspace/nacl_files/nacl_files_source.test.ts:210]: this test title basically says `getParsedNaclFile` should always return undefined, which, according to the next test, is not true
maybe add some description about when it should return undefined?

File: packages/workspace/test/common/nacl_file_source.ts (hunk 2 above)
Comment (amitgoldarb) [reply packages/workspace/test/common/nacl_file_source.ts:60]: Fixed

File: packages/workspace/test/workspace/nacl_files/nacl_files_source.test.ts (hunk 3 above)
Comment (amitgoldarb) [reply packages/workspace/test/workspace/nacl_files/nacl_files_source.test.ts:210]: Fixed

File: packages/workspace/src/workspace/nacl_files/nacl_files_source.ts (hunk 1 above)
Comment (amitgoldarb) [reply packages/workspace/src/workspace/nacl_files/nacl_files_source.ts]: Fixed

File: packages/workspace/src/workspace/nacl_files/nacl_files_source.ts [hunk 4]
```diff
@@ -369,10 +371,23 @@ const buildNaclFilesSource = (
...
       _.sum(await Promise.all([naclFilesStore.getTotalSize(), staticFileSource.getTotalSize()])),
 
-    getNaclFile: filename => naclFilesStore.get(filename),
+    getNaclFile,
 
-    getElements: async filename =>
-      Object.values((await getState()).parsedNaclFiles[filename]?.elements) || [],
+    getParsedNaclFile: async filename => {
+      // We don't want to parse all nacl files here when we want only parsedResult of one file.
+      if (state) {
```
Comment (ori-moisis) [inline packages/workspace/src/workspace/nacl_files/nacl_files_source.ts:378]: nit:
```suggestion
      if (state !== undefined) {
```

File: packages/workspace/src/workspace/nacl_files/nacl_files_source.ts [hunk 5]
```diff
@@ -369,10 +371,23 @@ const buildNaclFilesSource = (
...
+        return (await getState()).parsedNaclFiles[filename]
+      }
+      const naclFile = await getNaclFile(filename)
+      if (!naclFile) return undefined
```
Comment (ori-moisis) [inline packages/workspace/src/workspace/nacl_files/nacl_files_source.ts:382]: nit:
```suggestion
      if (naclFile === undefined) return undefined
```

File: packages/workspace/src/workspace/nacl_files/nacl_files_source.ts [hunk 6]
```diff
@@ -369,10 +371,23 @@ const buildNaclFilesSource = (
...
+      const cachedResult = await cache.get(cacheResultKey(naclFile.filename, naclFile.timestamp))
+      const parseResult = cachedResult || await parse(
+        Buffer.from(naclFile),
+        filename,
+        functions
+      )
+      return toParsedNaclFile(naclFile, parseResult)
```
Comment (ori-moisis) [inline packages/workspace/src/workspace/nacl_files/nacl_files_source.ts:389]: why re-write the cache logic and not use `parseNaclFiles`?

File: packages/workspace/src/workspace/nacl_files/nacl_files_source.ts (hunk 4 above)
Comment (amitgoldarb) [reply packages/workspace/src/workspace/nacl_files/nacl_files_source.ts:378]: Fixed

File: packages/workspace/src/workspace/nacl_files/nacl_files_source.ts (hunk 5 above)
Comment (amitgoldarb) [reply packages/workspace/src/workspace/nacl_files/nacl_files_source.ts:382]: Fixed

File: packages/workspace/src/workspace/nacl_files/nacl_files_source.ts (hunk 6 above)
Comment (amitgoldarb) [reply packages/workspace/src/workspace/nacl_files/nacl_files_source.ts]: Fixed

**Review by ori-moisis — APPROVED** [review]

**Pull request merged**